Automattic/woocommerce-payments
# Add button rules to appearance

Fixes https://github.com/Automattic/woopay/issues/2812

#### Changes proposed in this Pull Request
This PR adds button rules to appearance to allow us to get button styles from checkout, cart and product pages.

EPIC https://github.com/Automattic/woopay/issues/2781
SPIKE [here](https://woopayp2.wordpress.com/2024/07/01/spike-global-themes-support/#proof-of-concept-for-option-3-preferred)
PT [here](https://woopayp2.wordpress.com/2024/07/11/project-thread-woopay-global-themes-support/)

<img width="971" alt="button var" src="https://github.com/user-attachments/assets/9e917ecc-1e1d-4c48-a992-62ed48923c8a">





#### Testing instructions
* Test with [WooPay PR](https://github.com/Automattic/woopay/pull/2839) to see buttons styles applied on WooPay checkout page
* Go to WooPay admin -> WCPay Dev and enable WooPay global theme support feature flag

Test 1 ( The latest WooPayments version ):
1. Test this PR with the latest WooPayments version that includes `getAppearance` on the merchant site
1. Open dev tools
2. Initialize WooPay on the Blocks Checkout/ Classic Checkout / Blocks Cart / Classic Cart / Product page
3. Find `get_woopay_session` request and find `appearance[rules][.Button]`


Test 2 ( An older WooPayments version ):
1. Test an older version of WooPayments like [7.6.0](https://github.com/Automattic/woocommerce-payments/releases/download/7.6.0/woocommerce-payments.zip) that doesn't have `getAppearance` on the merchant site for regression test
2. Repeat the above testing instructions



-------------------


**Post merge**
